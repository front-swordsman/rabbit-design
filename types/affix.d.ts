import { ReactNode } from 'react'

export interface AffixProps {
  // 距离窗口顶部达到指定偏移量后触发
  offsetTop?: number
  // 距离窗口底部达到指定偏移量后触发
  offsetBottom?: number
  // 子元素
  children?: ReactNode | String
  // 容器
  target?: () => HTMLElement
  // 固定状态改变时触发的回调函数
  onChange?: (affixed: boolean) => void
  /* 样式相关 */
  className?: string
  // 样式
  style?: React.CSSProperties
}
