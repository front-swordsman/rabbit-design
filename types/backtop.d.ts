export interface BacktopProps {
  /* 滚动高度达到此参数值才出现 BackTop */
  visibilityHeight?: number
  /* 传入子节点 */
  children?: React.ReactNode
  /* 样式相关 */
  className?: string
  /* 样式 */
  style?: React.CSSProperties
}
