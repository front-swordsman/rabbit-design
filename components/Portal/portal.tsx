import * as React from 'react'
import * as ReactDOM from 'react-dom'
import { CSSTransition } from 'react-transition-group'
import cx from 'classnames'
import { Portal } from 'types/portal'
import { useClickOutSide, useEnhancedEffect } from '../utils/use'
import { noop } from '../utils/tool'

const defaultProps = {
  trigger: 'hover',
  placement: 'bottomLeft',
  disabled: false,
  wrapperComponent: 'span',
  prefixCls: 'snake-dropdown',
  hasTriangle: false,
  timeout: 300,
  offset: 0,
  mode: 'dropdown',
  visible: false,
  onVisibleChange: noop,
  destroy: true
}

const PortalOverlay: React.FC<Portal> = dropdownProps => {
  const props = { ...defaultProps, ...dropdownProps }
  const {
    className,
    placement,
    content,
    wrapperComponent: Component,
    children,
    trigger,
    disabled,
    prefixCls,
    animationName,
    wrapperStyle,
    style,
    onVisibleChange,
    hasTriangle,
    timeout,
    offset,
    mode,
    visible,
    destroy,
    getPopupContainer
  } = props
  const wrapperRef = React.useRef<HTMLElement>()
  const contentRef = React.useRef<HTMLDivElement>()
  const timeoutRef = React.useRef<any>()
  const [mountNode, setMountNode] = React.useState(null)

  useClickOutSide([contentRef, wrapperRef], () => {
    onVisibleChange(false)
  })

  const toggle = (triggerStr: string, toggle: boolean) => {
    if (disabled) return
    if (trigger === triggerStr) {
      // 只在变化的时候回调
      if (toggle !== visible) {
        onVisibleChange(toggle)
      }
    }
  }

  const getContainer = () => {
    if (getPopupContainer) {
      return getPopupContainer()
    }
    return document.body
  }

  const getOverlayClass = React.useCallback(() => {
    if (animationName) return `${prefixCls}-${animationName}`
    return `${prefixCls}-slide`
  }, [animationName])

  const getContentStyle = (top: string, left: string) => {
    ;(contentRef.current as HTMLElement).style.top = top
    ;(contentRef.current as HTMLElement).style.left = left
  }

  const getPosition = () => {
    const rect = (wrapperRef.current as HTMLElement).getBoundingClientRect()
    const contentRect = (contentRef.current as HTMLElement).getBoundingClientRect()
    let top = `${rect.top + rect.height + offset + window.pageYOffset}px`
    let left = `${rect.left + window.pageXOffset}px`
    if (contentRect.width < rect.width && mode === 'dropdown') {
      contentRef.current.style.minWidth = `${rect.width}px`
    }
    switch (placement) {
      case 'top':
        top = `${rect.top - offset - contentRect.height + window.pageYOffset}px`
        left = `${rect.left + rect.width / 2 - contentRect.width / 2 + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'topLeft':
        top = `${rect.top - offset - contentRect.height + window.pageYOffset}px`
        left = `${rect.left + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'topRight':
        top = `${rect.top - offset - contentRect.height + window.pageYOffset}px`
        left = `${rect.right - contentRect.width + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'bottom':
        top = `${rect.top + rect.height + offset + window.pageYOffset}px`
        left = `${rect.left + rect.width / 2 - contentRect.width / 2 + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'bottomLeft':
        top = `${rect.top + rect.height + offset + window.pageYOffset}px`
        left = `${rect.left + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'bottomRight':
        top = `${rect.top + rect.height + offset + window.pageYOffset}px`
        left = `${rect.right - contentRect.width + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'left':
        top = `${rect.top + rect.height / 2 - contentRect.height / 2 + window.pageYOffset}px`
        left = `${rect.left - offset - contentRect.width + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'leftTop':
        top = `${rect.top + window.pageYOffset}px`
        left = `${rect.left - offset - contentRect.width + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'leftBottom':
        top = `${rect.top + rect.height - contentRect.height + window.pageYOffset}px`
        left = `${rect.left - offset - contentRect.width + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'right':
        top = `${rect.top + rect.height / 2 - contentRect.height / 2 + window.pageYOffset}px`
        left = `${rect.right + offset + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'rightTop':
        top = `${rect.top + window.pageYOffset}px`
        left = `${rect.right + offset + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'rightBottom':
        top = `${rect.top + rect.height - contentRect.height + window.pageYOffset}px`
        left = `${rect.right + offset + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      default:
        getContentStyle(top, left)
        break
    }
  }

  const clearTime = () => {
    if (timeoutRef.current) {
      clearTimeout(timeoutRef.current)
    }
  }

  const handleClick = () => {
    toggle('click', true)
  }

  const handleMouseEnter = () => {
    clearTime()
    timeoutRef.current = setTimeout(() => {
      toggle('hover', true)
    }, 0)
  }

  const handleMouseLeave = () => {
    clearTime()
    timeoutRef.current = setTimeout(() => {
      toggle('hover', false)
    }, 0)
  }

  const renderDropDown = () => {
    const classStr = cx(
      prefixCls,
      {
        [`${prefixCls}-${placement}`]: placement,
        [`${prefixCls}-triangle`]: hasTriangle
      },
      className
    )
    return (
      <CSSTransition
        in={visible}
        timeout={timeout}
        classNames={getOverlayClass()}
        unmountOnExit={destroy}
      >
        <div
          className={classStr}
          ref={contentRef}
          onClick={() => {
            onVisibleChange(false)
          }}
          onMouseEnter={handleMouseEnter}
          onMouseLeave={handleMouseLeave}
          style={style}
        >
          <div className={`${prefixCls}-content`}>{content}</div>
        </div>
      </CSSTransition>
    )
  }

  React.useEffect(() => {
    setMountNode(getContainer())
  }, [getContainer()])

  useEnhancedEffect(() => {
    if (visible && mountNode) {
      getPosition()
    }
  }, [visible, mountNode])

  return (
    <>
      <Component
        onClick={handleClick}
        onMouseEnter={handleMouseEnter}
        onMouseLeave={handleMouseLeave}
        style={wrapperStyle}
        ref={wrapperRef}
      >
        {React.cloneElement(children as any, {
          disabled: disabled
        })}
      </Component>
      {mountNode ? ReactDOM.createPortal(renderDropDown(), getContainer()) : null}
    </>
  )
}

export default PortalOverlay
