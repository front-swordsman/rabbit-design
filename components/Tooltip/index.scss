@import '../styles/themes/default';

$tooltip: 'snake-tooltip';

.#{$tooltip} {
  position: absolute;
  z-index: $zindexPopover;
  box-sizing: border-box;
  color: rgba(0, 0, 0, 0.65);
  font-size: $body;
  line-height: $lineHeight;
  list-style: none;
  cursor: pointer;
  margin: 0;
  text-align: left;
  visibility: visible;
  max-width: 250px;

  &-content {
    box-shadow: 0 2px 8px rgba(0, 0, 0, 0.15);
    border-radius: $borderRadius;
  }

  &-inner {
    min-width: 30px;
    padding: 6px 8px;
    color: #fff;
    text-align: left;
    text-decoration: none;
    word-wrap: break-word;
    background-color: rgba(0, 0, 0, 0.75);
    border-radius: 4px;
    box-shadow: 0 2px 8px rgba(0, 0, 0, 0.15);
    box-sizing: border-box;
  }

  &-top {
    transform-origin: 50% 100%;
  }

  &-topLeft {
    transform-origin: 0 100%;
  }

  &-topRight {
    transform-origin: 100% 100%;
  }

  &-bottom {
    transform-origin: 50% 0;
  }

  &-bottomLeft {
    transform-origin: 0 0;
  }

  &-bottomRight {
    transform-origin: 100% 0;
  }

  &-left {
    transform-origin: 100% 50%;
  }

  &-leftTop {
    transform-origin: 100% 0;
  }

  &-leftBottom {
    transform-origin: 100% 100%;
  }

  &-right {
    transform-origin: 0 50%;
  }

  &-rightTop {
    transform-origin: 0 0;
  }

  &-rightBottom {
    transform-origin: 0 100%;
  }

  &-triangle {
    .#{$tooltip}-content {
      &::after {
        content: '';
        position: absolute;
        width: 0;
        height: 0;
        border-color: transparent;
        border-style: solid;
        box-sizing: border-box;
      }
    }
  }

  &-triangle.#{$tooltip}-top,
  &-triangle.#{$tooltip}-topLeft,
  &-triangle.#{$tooltip}-topRight {
    .#{$tooltip}-content {
      &::after {
        bottom: 5px;
        border-width: 5px 5px 0;
        border-top-color: rgba(0, 0, 0, 0.75);
      }
    }
  }

  &-triangle.#{$tooltip}-top,
  &-triangle.#{$tooltip}-bottom {
    .#{$tooltip}-content {
      &::after {
        left: 50%;
        transform: translateX(-50%);
      }
    }
  }

  &-triangle.#{$tooltip}-topLeft,
  &-triangle.#{$tooltip}-bottomLeft {
    .#{$tooltip}-content {
      &::after {
        left: 16px;
      }
    }
  }

  &-triangle.#{$tooltip}-topRight,
  &-triangle.#{$tooltip}-bottomRight {
    .#{$tooltip}-content {
      &::after {
        right: 16px;
      }
    }
  }

  &-triangle.#{$tooltip}-bottom,
  &-triangle.#{$tooltip}-bottomLeft,
  &-triangle.#{$tooltip}-bottomRight {
    .#{$tooltip}-content {
      &::after {
        top: 5px;
        border-width: 0 5px 5px;
        border-bottom-color: rgba(0, 0, 0, 0.75);
      }
    }
  }

  &-triangle.#{$tooltip}-left,
  &-triangle.#{$tooltip}-leftTop,
  &-triangle.#{$tooltip}-leftBottom {
    .#{$tooltip}-content {
      &::after {
        right: 5px;
        border-width: 5px 0 5px 5px;
        border-left-color: rgba(0, 0, 0, 0.75);
      }
    }
  }

  &-triangle.#{$tooltip}-right,
  &-triangle.#{$tooltip}-rightTop,
  &-triangle.#{$tooltip}-rightBottom {
    .#{$tooltip}-content {
      &::after {
        left: 5px;
        border-width: 5px 5px 5px 0;
        border-right-color: rgba(0, 0, 0, 0.75);
      }
    }
  }

  &-triangle.#{$tooltip}-leftTop,
  &-triangle.#{$tooltip}-rightTop {
    .#{$tooltip}-content {
      &::after {
        top: 8px;
      }
    }
  }

  &-triangle.#{$tooltip}-left,
  &-triangle.#{$tooltip}-right {
    .#{$tooltip}-content {
      &::after {
        top: 50%;
        // transform: translateY(-50%);
        margin-top: -5px;
      }
    }
  }

  &-triangle.#{$tooltip}-leftBottom,
  &-triangle.#{$tooltip}-rightBottom {
    .#{$tooltip}-content {
      &::after {
        bottom: 8px;
      }
    }
  }

  &-top,
  &-topLeft,
  &-topRight {
    padding-bottom: 10px;
  }

  &-bottom,
  &-bottomLeft,
  &-bottomRight {
    padding-top: 10px;
  }

  &-left,
  &-leftTop,
  &-leftBottom {
    padding-right: 10px;
  }

  &-right,
  &-rightTop,
  &-rightBottom {
    padding-left: 10px;
  }

  &-tooltip-animate-enter,
  &-tooltip-animate-appear {
    animation-duration: 0.2s;
    animation-fill-mode: both;
    animation-play-state: paused;
  }

  &-tooltip-animate-enter-active,
  &-tooltip-animate-appear-active {
    animation-duration: 0.2s;
    animation-fill-mode: both;
    animation-name: ZoomEnter;
    animation-play-state: running;
  }

  &-tooltip-animate-exit {
    animation-duration: 0.2s;
    animation-fill-mode: both;
    animation-play-state: paused;
  }

  &-tooltip-animate-exit-active {
    animation-name: ZoomLeave;
    animation-play-state: running;
    pointer-events: none;
  }
}

@keyframes ZoomEnter {
  0% {
    transform: scale(0.8);
    opacity: 0;
  }

  100% {
    transform: scale(1);
    opacity: 1;
  }
}

@keyframes ZoomLeave {
  0% {
    transform: scale(1);
  }

  100% {
    transform: scale(0.8);
    opacity: 0;
  }
}
