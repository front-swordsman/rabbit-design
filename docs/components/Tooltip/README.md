---
imports: 
  import BasicDemoCode from '!raw-loader!./basic.tsx';
  import BasicDemo from './basic.tsx';
  import SimpleDemoCode from '!raw-loader!./simpleDemo.tsx';
  import SimpleDemo from './simpleDemo.tsx';
---

# Tooltip 文字提示

简单的文字提示气泡框。

## 基础用法

:::demo
<Block code={BasicDemoCode} des="基础用法">
<BasicDemo />
</Block>
:::

## 多个方向

:::demo
<Block code={SimpleDemoCode} des="简单用法">
<SimpleDemo />
</Block>
:::