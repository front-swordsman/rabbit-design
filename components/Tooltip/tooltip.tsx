import * as React from 'react'
import { TooltipProps } from 'types/tooltip'
import Portal from '../Portal'
import { noop } from '../utils/tool'

import './index.scss'

const defaultProps: Partial<TooltipProps> = {
  placement: 'top',
  visible: false,
  onVisibleChange: noop,
  trigger: 'hover',
  autoAdjustOverflow: true
}

const prefixCls = 'snake-tooltip'

export default function Tooltip(tooltip: TooltipProps) {
  const props = { ...defaultProps, ...tooltip }
  const { title, contentClass, contentStyle, ...rest } = props

  const renderContent = () => {
    return (
      <div className={`${prefixCls}-inner`} role="tooltip">
        {title ? <div className={`${prefixCls}-inner-title`}>{title}</div> : null}
      </div>
    )
  }
  return (
    <Portal
      prefixCls={prefixCls}
      mode="tooltip"
      content={renderContent()}
      hasTriangle
      className={contentClass}
      style={contentStyle}
      animationName="tooltip-animate"
      {...rest}
    />
  )
}
