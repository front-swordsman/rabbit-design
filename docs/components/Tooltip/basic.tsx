import * as React from 'react'
import { Tooltip, Button } from 'components'

export default function SimpleDemo() {
  const [visible, setVisible] = React.useState(false)
  return (
    <div style={{ marginTop: 20 }}>
      <Tooltip
        visible={visible}
        onVisibleChange={visible => setVisible(visible)}
        title="内容区"
        wrapperComponent="div"
        wrapperStyle={{ display: 'inline-block' }}
        trigger="click"
      >
        <Button>Hover me</Button>
      </Tooltip>
    </div>
  )
}
