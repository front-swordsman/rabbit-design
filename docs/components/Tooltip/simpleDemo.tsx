import * as React from 'react'
import { Tooltip, Button, Radio } from 'components'

export default function SimpleDemo() {
  const [visible, setVisible] = React.useState(false)
  const [direction, setDirection] = React.useState<any>('top')
  const [action, setAction] = React.useState<any>('click')
  return (
    <div style={{ marginTop: 20 }}>
      <Tooltip
        visible={visible}
        onVisibleChange={visible => setVisible(visible)}
        title="内容内容区内容"
        wrapperComponent="div"
        wrapperStyle={{ display: 'inline-block' }}
        trigger={action}
        placement={direction}
      >
        <Button>点击我</Button>
      </Tooltip>
      <div style={{ marginTop: 100 }}>
        <Radio
          options={[
            { label: 'bottomLeft', value: 'bottomLeft' },
            { label: 'bottom', value: 'bottom' },
            { label: 'bottomRight', value: 'bottomRight' },
            { label: 'topLeft', value: 'topLeft' },
            { label: 'top', value: 'top' },
            { label: 'topRight', value: 'topRight' },
            { label: 'left', value: 'left' },
            { label: 'leftTop', value: 'leftTop' },
            { label: 'leftBottom', value: 'leftBottom' },
            { label: 'right', value: 'right' },
            { label: 'rightTop', value: 'rightTop' },
            { label: 'rightBottom', value: 'rightBottom' }
          ]}
          value={direction}
          onChange={(value: string) => setDirection(value)}
        />
      </div>
      <div>
        <Radio
          options={[{ label: 'click', value: 'click' }, { label: 'hover', value: 'hover' }]}
          value={action}
          onChange={(value: string) => setAction(value)}
        />
      </div>
    </div>
  )
}
