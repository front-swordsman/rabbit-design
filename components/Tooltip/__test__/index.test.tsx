import * as React from 'react'
import { render } from 'react-testing-library'
import Tooltip from '../index'
import { Placement } from 'types/portal'

jest.mock('react-transition-group', () => {
  return {
    CSSTransition: props => {
      return <div>{props.in || !props.unmountOnExit ? props.children : null}</div>
    }
  }
})

describe('Tooltip Test', () => {
  function PopoverDemo({
    placement = 'top',
    visible = true
  }: {
    placement?: Placement
    visible?: boolean
  }) {
    return (
      <Tooltip placement={placement} visible={visible} title={<div>内容区</div>}>
        <span>click me</span>
      </Tooltip>
    )
  }

  it('snapshot when visible is false', () => {
    const { baseElement } = render(<PopoverDemo visible={false} />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot when visible is true', () => {
    const { baseElement } = render(<PopoverDemo />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot with placement is topLeft', () => {
    const { baseElement } = render(<PopoverDemo placement="topLeft" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot with placement is topRight', () => {
    const { baseElement } = render(<PopoverDemo placement="topRight" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot with placement is bottom', () => {
    const { baseElement } = render(<PopoverDemo placement="bottom" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot with placement is bottomLeft', () => {
    const { baseElement } = render(<PopoverDemo placement="bottomLeft" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot with placement is bottomRight', () => {
    const { baseElement } = render(<PopoverDemo placement="bottomRight" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot with placement is left', () => {
    const { baseElement } = render(<PopoverDemo placement="left" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot with placement is leftTop', () => {
    const { baseElement } = render(<PopoverDemo placement="leftTop" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot with placement is leftBottom', () => {
    const { baseElement } = render(<PopoverDemo placement="leftBottom" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot with placement is right', () => {
    const { baseElement } = render(<PopoverDemo placement="right" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot with placement is rightTop', () => {
    const { baseElement } = render(<PopoverDemo placement="rightTop" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot with placement is rightBottom', () => {
    const { baseElement } = render(<PopoverDemo placement="rightBottom" />)
    expect(baseElement).toMatchSnapshot()
  })
})
