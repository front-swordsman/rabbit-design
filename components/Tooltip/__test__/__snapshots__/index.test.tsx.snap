// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Test snapshot when visible is false 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div />
</body>
`;

exports[`Tooltip Test snapshot when visible is true 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-tooltip snake-tooltip-top snake-tooltip-triangle"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-tooltip-content"
      >
        <div
          class="snake-tooltip-inner"
          role="tooltip"
        >
          <div
            class="snake-tooltip-inner-title"
          >
            <div>
              内容区
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Tooltip Test snapshot with placement is bottom 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-tooltip snake-tooltip-bottom snake-tooltip-triangle"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-tooltip-content"
      >
        <div
          class="snake-tooltip-inner"
          role="tooltip"
        >
          <div
            class="snake-tooltip-inner-title"
          >
            <div>
              内容区
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Tooltip Test snapshot with placement is bottomLeft 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-tooltip snake-tooltip-bottomLeft snake-tooltip-triangle"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-tooltip-content"
      >
        <div
          class="snake-tooltip-inner"
          role="tooltip"
        >
          <div
            class="snake-tooltip-inner-title"
          >
            <div>
              内容区
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Tooltip Test snapshot with placement is bottomRight 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-tooltip snake-tooltip-bottomRight snake-tooltip-triangle"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-tooltip-content"
      >
        <div
          class="snake-tooltip-inner"
          role="tooltip"
        >
          <div
            class="snake-tooltip-inner-title"
          >
            <div>
              内容区
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Tooltip Test snapshot with placement is left 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-tooltip snake-tooltip-left snake-tooltip-triangle"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-tooltip-content"
      >
        <div
          class="snake-tooltip-inner"
          role="tooltip"
        >
          <div
            class="snake-tooltip-inner-title"
          >
            <div>
              内容区
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Tooltip Test snapshot with placement is leftBottom 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-tooltip snake-tooltip-leftBottom snake-tooltip-triangle"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-tooltip-content"
      >
        <div
          class="snake-tooltip-inner"
          role="tooltip"
        >
          <div
            class="snake-tooltip-inner-title"
          >
            <div>
              内容区
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Tooltip Test snapshot with placement is leftTop 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-tooltip snake-tooltip-leftTop snake-tooltip-triangle"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-tooltip-content"
      >
        <div
          class="snake-tooltip-inner"
          role="tooltip"
        >
          <div
            class="snake-tooltip-inner-title"
          >
            <div>
              内容区
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Tooltip Test snapshot with placement is right 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-tooltip snake-tooltip-right snake-tooltip-triangle"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-tooltip-content"
      >
        <div
          class="snake-tooltip-inner"
          role="tooltip"
        >
          <div
            class="snake-tooltip-inner-title"
          >
            <div>
              内容区
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Tooltip Test snapshot with placement is rightBottom 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-tooltip snake-tooltip-rightBottom snake-tooltip-triangle"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-tooltip-content"
      >
        <div
          class="snake-tooltip-inner"
          role="tooltip"
        >
          <div
            class="snake-tooltip-inner-title"
          >
            <div>
              内容区
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Tooltip Test snapshot with placement is rightTop 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-tooltip snake-tooltip-rightTop snake-tooltip-triangle"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-tooltip-content"
      >
        <div
          class="snake-tooltip-inner"
          role="tooltip"
        >
          <div
            class="snake-tooltip-inner-title"
          >
            <div>
              内容区
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Tooltip Test snapshot with placement is topLeft 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-tooltip snake-tooltip-topLeft snake-tooltip-triangle"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-tooltip-content"
      >
        <div
          class="snake-tooltip-inner"
          role="tooltip"
        >
          <div
            class="snake-tooltip-inner-title"
          >
            <div>
              内容区
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Tooltip Test snapshot with placement is topRight 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-tooltip snake-tooltip-topRight snake-tooltip-triangle"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-tooltip-content"
      >
        <div
          class="snake-tooltip-inner"
          role="tooltip"
        >
          <div
            class="snake-tooltip-inner-title"
          >
            <div>
              内容区
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
